pkg/controller/controller.go

```diff
@@ -110,26 +110,45 @@ type controller struct {
 	brokerLister           listers.BrokerLister
 	serviceClassLister     listers.ServiceClassLister
 	instanceLister         listers.InstanceLister
+	bindingLister          listers.BindingLister
+	brokerQueue            workqueue.RateLimitingInterface
+	serviceClassQueue      workqueue.RateLimitingInterface
+	instanceQueue          workqueue.RateLimitingInterface
+	bindingQueue           workqueue.RateLimitingInterface
 }
 
 // Run runs the controller until the given stop channel can be read from.
-func (c *controller) Run(stopCh <-chan struct{}) {
+func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 	defer runtimeutil.HandleCrash()
+	defer func() {
+		c.brokerQueue.ShutDown()
+		c.serviceClassQueue.ShutDown()
+		c.instanceQueue.ShutDown()
+		c.bindingQueue.ShutDown()
+	}()
+
 	glog.Info("Starting service-catalog controller")
 
+	for i := 0; i < workers; i++ {
+		go wait.Until(c.brokerWorker, time.Second, stopCh)
+		go wait.Until(c.serviceClassWorker, time.Second, stopCh)
+		go wait.Until(c.instanceWorker, time.Second, stopCh)
+		go wait.Until(c.bindingWorker, time.Second, stopCh)
```

Comment:
It isn't a timeout; it is a retry interval if the function happens to return; which it shouldn't unless it.  Also, if `stopCh` is closed, wait.Until() returns and thus the whole goroutine exits.

The comment would really belong in the godoc for `wait.Until()`.